zap/src/test/java/org/zaproxy/zap/utils/XmlUtilsUnitTest.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2025 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.utils;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import org.junit.jupiter.api.Test;
+
+class XmlUtilsUnitTest {
+
+    @Test
+    void shouldReturnStrForEmptyXml() {
+        // given
+        String xml = "<aaa/>";
+        // when
+        String result = XmlUtils.getXmlKeyString(xml);
+        // then
+        assertThat(result, is("<aaa>"));
+    }
+
+    @Test
+    void shouldReturnStrForSimpleXml() {
+        // given
+        String xml = "<aaa><bbb>BBB</bbb><ccc>CCC</ccc><ddd>DDD</ddd></aaa>";
+        // when
+        String result = XmlUtils.getXmlKeyString(xml);
+        // then
+        assertThat(result, is("<aaa:<bbb>,<ccc>,<ddd>>"));
+    }
+
+    @Test
+    void shouldReturnStrForComplexElementXml() {
+        // given
+        String xml =
+                "<description>It happened on <date lang=\"norwegian\">03.03.99</date> etc</description> ";
+        // when
+        String result = XmlUtils.getXmlKeyString(xml);
+        // then
+        assertThat(result, is("<description:<date>>"));
+    }
+
+    @Test
+    void shouldReturnStrForDeeperDuplicatedXml() {
+        // given
+        String xml =
+                """
```

Comment:
Oh here too :)

Reply:
For longer blocks of text I dont think that makes sense. And spotlessApply will probably just change them back anyway 😛